Lake Lighting Project

Hazado opened the pull request:
Number of changes made to the water and Cave Backdrop in each age.
Layer Animations added for 5 states of brightness for water and backdrop.
Dark, Quarter, Half, ThreeQuarter, Full

Layer animations are set to a full Dni Day and cause the water to dim and glow from midnight to noon to midnight (Dni equivalents).

Some lighting may be too bright, used the lighting from the Neighborhood as this was the only place this was implemented. We are able to dim the lighting by changing the x/y coords in the layer animation.

I had to do this for the city due to how radioactive orange it was.

https://github.com/H-uru/Plasma/pull/938

Hazado commented:
Still need to do Great Zero

Fix the gap between the Arch in city

Hazado commented:
Great Zero done.
Might have the various areas use the larger dnicitybackdropnoarch_0#0.hsm texture
Should make the backdrop look a little better

Hoikas commented:
Would you mind working with @DoobesURU to get the Vothol version in the Vothol blend file? I'd also appreciate a description of which PRP objects you added or changed (and how) in the commit history in case we ever have to replicate these changes.

DoobesURU commented:
I figure the easiest way to do this would be:

a) I get a checklist of all the changes and I can translate them into Blender/Korman, or
b) I can have a look at the modified PRPs and translate _that_ into the blend.

That way, we can have an exportable copy for anyone who wants to make further modifications down the line. AFAIK, all the changes can be done in Korman as well.

Hazado commented:
I can get you a list probably later today or tomorrow. Need to do an xml comparison first.

Hazado commented:
Vothol Changes
Added to Modifiers "PythLakeLightCycle0" Python File mod to Scene Object "WaterMirrorPlane"
Added to Modifiers "PythLakeLightCycle" Python File mod to Scene Object "Water"
Changed Ambient color of "water0_Map__2311" Layer to red="0.5607843399" green="0.2745098174" blue="0.01568627544" from red="0" green="0" blue="0"
Changed "Water2" Material Layer to "Map #239_anim_lakeLightDark" and checked "Needs Blend Channel"
Changed "water0_Map__2311_LayerAnim" Layer Animation base layer to "Map #2311_anim_lakeLightDark"
Added Total of 10 Layer Animations (Dark, Quarter, Half, ThreeQuarter, Full) for Map #2311 and Map #239
Added 2 Python File mods "PythLakeLightCycle" and "PythLakeLightCycle0"

EhrenCG commented:
Oh, even involving the Baron City Office?  Do you know about its fog problem?
Should add this line to BaronCityOffice's fni file:

Graphics.Renderer.Fog.SetDefLinear -300 3000 1

As it stands now, the look of things outside the office windows varies widely depending on if you link to Sharper's Office from Teledahn or from Relto.  Ae'gura seems to look much more natural linking in from Teledahn, so that line is copied from Teledahn's fni file.

Since the nature of what you're doing involves brightness, this seems relevant to avoid confusion about light levels.

Hazado commented:
5 new backdrops added to the city, these were created by Emor
City Backdrop mesh was also redone by Emor, should eliminate the strange curving and lines in the sky

Deledrius commented:
> 
> 
> 5 new backdrops added to the city, these were created by Emor City Backdrop mesh was also redone by Emor, should eliminate the strange curving and lines in the sky

Awesome!  Glad this is being added/fixed up! :D

EhrenCG commented:
> 
> 
> 5 new backdrops added to the city, these were created by Emor City Backdrop mesh was also redone by Emor, should eliminate the strange curving and lines in the sky

Whoa, hang on!  I'm like 99% certain those curving lines in the sky are SUPPOSED to be there, they make it look like the cave ceiling.  I always thought they looked good... hmm.

Emor-Dni-Lap commented:
> Whoa, hang on! I'm like 99% certain those curving lines in the sky are SUPPOSED to be there, they make it look like the cave ceiling. I always thought they looked good... hmm.
Hi Ehren - those streaks were literally the upper parts of the city-backdrop art **dnicitybackdropnoarch_40_1** (buildings and rock spires) that had been mis-UVed into the upper part of a squashed-sphere skydome.
And that squashed sphere was also the reason the city art appeared curved. What I did was to create a proper cylindrical backdrop, permitting all vertical buildings etc. to remain vertical, and re-UVed to keep the sky black in this version.
Hazado and I have talked about what to do about the 'ceiling' as the lake lighting gets brighter and brighter, but have agreed that if something is done, it should be as a separate element. Whether we treat it as just texture, or create a true mesh with stalactites (in which case they would pierce the animated 'fog' layer) is open for discussion.

EhrenCG commented:
> 
> 
> @EhrenCG wrote: ` Whoa, hang on! I'm like 99% certain those curving lines in the sky are SUPPOSED to be there, they make it look like the cave ceiling. I always thought they looked good... hmm.` Hi Ehren - those streaks were literally the upper parts of the city-backdrop art **dnicitybackdropnoarch_40_1** (buildings and rock spires) that had been mis-UVed into the upper part of a squashed-sphere skydome. And that squashed sphere was also the reason the city art appeared curved. What I did was to create a proper cylindrical backdrop, permitting all vertical buildings etc. to remain vertical, and re-UVed to keep the sky black in this version. Hazado and I have talked about what to do about the 'ceiling' as the lake lighting gets brighter and brighter, but have agreed that if something is done, it should be as a separate element. Whether we treat it as just texture, or create a true mesh with stalactites (in which case they would pierce the animated 'fog' layer) is open for discussion.

Ugh, it would help so much if I could figure out these large file size things from git hub to see it for myself, but I can't find anything that actually helps with the program for doing that.  I don't know how to use the command prompt to download something in a repository of Git Hub off of the internet, and I feel like even when I tried to do some commands I got "git" being an unrecognized command or something, I so need a plain step by step explanation or I'll always be at risk of misunderstanding what's being referenced as changed... (I'm kind of unsure we are talking about the same thing now to be honest).

If anyone could help with how to do this it would be insanely appreciated, unless it's just way too many steps to explain...  But I am already pretty discouraged by the results I've tried to find googling on my own.

EhrenCG commented:
> 
> 
> @EhrenCG wrote: ` Whoa, hang on! I'm like 99% certain those curving lines in the sky are SUPPOSED to be there, they make it look like the cave ceiling. I always thought they looked good... hmm.` Hi Ehren - those streaks were literally the upper parts of the city-backdrop art **dnicitybackdropnoarch_40_1** (buildings and rock spires) that had been mis-UVed into the upper part of a squashed-sphere skydome. And that squashed sphere was also the reason the city art appeared curved. What I did was to create a proper cylindrical backdrop, permitting all vertical buildings etc. to remain vertical, and re-UVed to keep the sky black in this version. Hazado and I have talked about what to do about the 'ceiling' as the lake lighting gets brighter and brighter, but have agreed that if something is done, it should be as a separate element. Whether we treat it as just texture, or create a true mesh with stalactites (in which case they would pierce the animated 'fog' layer) is open for discussion.

Sorry for a second post in a row, I deleted the other one since it was just me whining about LFS thing, but I thought to switch out with a brighter texture to see how things map (This is CC but I'm highly certain it's the same mapping in MOUL):
![MapCeiling](https://user-images.githubusercontent.com/10370754/140659618-ce479abf-ca80-40d4-a44b-e2ab3ca77e54.jpg)
I guess this is what the problem is.  Well, I still think it looked good as a dark ceiling through a layer of fog, since only the more bold lines in the distortion were noticeable to me, but I see the huge problem for if it were to be brighter.

Deledrius commented:
I agree that there is a need to have a semi-visible cavern ceiling, but I think @Emor-Dni-Lap is right that it would need to be done as its own thing, and cannot be the current mesh as-is.

It would be great to have some screenshots of the additions added here for review (and posterity).

> 
> 
> Sorry for a second post in a row, I deleted the other one since it was just me whining about LFS thing, 

I'd be happy to walk you through it.  LFS is a bit of a hurdle, unfortunately, but isn't too bad once you have it all set up.  I do wish there was a better solution.  If you drop by IRC sometime we can get you going (discord is too much of a firehose for me).

Hazado commented:
There are 5 levels of lighting, 0-4.
0 is dark with no changing lighting or fog
1 - 25% with fog and lighting change over the full d'ni day
2 - 50%
3 - 75%
4 - 100%

Midnight shots (Darkest)
0
![image](https://user-images.githubusercontent.com/416114/140660301-2a9fe98b-a141-4387-8a79-420fcf122507.png)
1
![image](https://user-images.githubusercontent.com/416114/140660359-cded245e-9a99-4498-9528-ef3af01c875f.png)
2
![image](https://user-images.githubusercontent.com/416114/140660363-e39655f2-f70a-4a76-98df-349d221bbf02.png)
3
![image](https://user-images.githubusercontent.com/416114/140660370-fed8d4f8-3a85-474d-b963-1763c024189d.png)
4
![image](https://user-images.githubusercontent.com/416114/140660375-c60345be-d9c3-4377-b539-38bb01f38f00.png)

Noon shots (Brightest)
1
![image](https://user-images.githubusercontent.com/416114/140660452-c9f47a88-1678-4001-bd4a-d8a8e66f0672.png)
2
![image](https://user-images.githubusercontent.com/416114/140660489-ae8bd4ac-8807-4451-832b-dc0ff36fdd58.png)
3
![image](https://user-images.githubusercontent.com/416114/140660513-a1224d0f-910d-4b09-b66c-f5c6711bc8ce.png)
4
![image](https://user-images.githubusercontent.com/416114/140660549-e32da396-fffc-4d35-9bd9-f616ec7bf2e4.png)

Deledrius commented:
Excellent, thank you!  That looks amazing.

Emor-Dni-Lap commented:
@EhrenCG -
You got it: the issue is the increased brightness, which not only amplifies what you've always seen in-cavern but makes a lot more garbage visible. And yeah, those streaks on the ceiling have always been just the unintended results of bad UVing in the first place...even though everyone has gotten so used to seeing them over the years.
Hopefully you'll be happier with a cavern ceiling created using more, um...proper...methods, which we may  implement in stages (still under discussion).

Emor-Dni-Lap commented:
@Hazado -
I hope you can adjust your brightness scalings for your 'midnight' and 'high noon' ranges such that the brightest level does not exceed that of the original Level4 art:
![image](https://user-images.githubusercontent.com/57068145/140958750-6c7cf3e4-8fd0-434f-aa8d-78079bef40db.png)

Hazado commented:
Fog max brightness is what is what would be normal for city before I made changes.

EhrenCG commented:
Ah, I finally got to see this on the Minkata Alpha shard!  However, I watched the great zero beam cross over the face of the arch and its animation has a slight skip backwards.  Had that been noticed yet?

Hazado commented:
> Ah, I finally got to see this on the Minkata Alpha shard! However, I watched the great zero beam cross over the face of the arch and its animation has a slight skip backwards. Had that been noticed yet?

This is actually a bug that exists in MOULA currently. Its pretty easy to fix, so ill try to get it in the next release. The animation is too long and the start point is behind the end point. Just need to shorten how long it plays to match them up better.

Emor-Dni-Lap commented:
> 
> 
> Fog max brightness is what is what would be normal for city before I made changes.

You're right, of course: I went in-game and the fog is way more dense than I remembered...it pretty much completely obscures almost all of the backdrop texture, which is also set so dark that little is seen of it. All this explains why no one has EVER commented on the backdrop repeating at least three times around Ae'gura, to my recollection:
![CityBackdrop_OriginalUVs](https://user-images.githubusercontent.com/57068145/141165232-ba881c4e-8ac4-4e94-874a-282d29746ae3.png)
And I think everything we're learning also goes a long way toward explaining why past Lake Lighting Projects never came to fruition: I'm betting that Cyan also came to the realization that they'd need new, more-detailed art, revised mesh and UVing, as well as changes in fog settings. Pretty much everything.

I'm at the point where I would advocate changing the fog settings a lot: to my eye at least, it's always seemed far brighter than would be appropriate to the low lake light levels we've always seen in-cavern. For our current lowest light level, it should barely be seen at all. And I'm no longer attached to the notion that it must be kept the way it is simply because X number of explorers say "that's what we're used to".

EhrenCG commented:
> I'm at the point where I would advocate changing the fog settings a lot: to my eye at least, it's always seemed far brighter than would be appropriate to the low lake light levels we've always seen in-cavern. For our current lowest light level, it should barely be seen at all. And I'm no longer attached to the notion that it must be kept the way it is simply because X number of explorers say "that's what we're used to".

We could start by trying changing the fog color from ".4 .3 .1" to ".3 .2 .05".  Not only is ".3 .2 .05" what they changed the city fog color to for CC, but it's also what even MOULa's other D'ni areas often use including K'veer, The Great Zero, and The Great Shaft.  So I would argue it's the closest to an "official confirmed" fog setting as we can get for something that's a change.  And it IS darker.

Not that I'm saying it should be an absolute constraint, just that I think it's a good place to start and see how it looks.

Deledrius commented:
> Not that I'm saying it should be an absolute constraint, just that I think it's a good place to start and see how it looks.

Sounds worth a try to me.

Emor-Dni-Lap commented:
As mentioned elsewhere to Hazado: the real problem with the current method for fog generation / rendering is that the fog has zero opacity, as real fog would. Instead, it renders in an additive fashion with anything in the background, effectively making a mess of any efforts to precisely determine a background texture's brightness. 
Like, exactly what we're trying to do here.

Emor-Dni-Lap commented:
Current evolution of our approach for the City backdrop art and mesh: 
- art width has been doubled to a 4X1 format to minimize visible tiling of the previous art (horizontal repeats that were previously unseen because the backdrop was so dark and obscured).
- Fog settings will be minimized in order to avoid light pollution on the backdrop image
- Lake lighting changes will instead be seen more prominently on the cloud layer
- Cloud layer has now been doubled with variant art to provide an evolving/morphing fog effect
- Cavern ceiling will now be a separate mesh with its own texture, to be seen only slightly at higher lake light levels

Though we're getting close to having a look we're satisfied with, we don't feel these changes can be brought into the Cyan shard until they are meaningful: in other words, until the brightness levels are driven by pellet drop counts. Since we don't think it would be either practical or fair to place  the responsibility for monitoring pellet counts and manually making changes in brightness settings on any one person, this process needs to be automated (with some manual intervention). Now, since we want to see the lake respond to pellet drops - but we'll never know in advance what the numbers will be - the manual intervention would be in the form of a fractional multiplier that could advance or retard the lake's response curve, depending on how fast or slowly pellet numbers are coming in. That way, there'd still be an accurate and automated system, but with a simple way to adjust the overall responsiveness.
This would require some code that would drive the SDL settings for lake / cloud / backdrop art, based on weekly updates in data files made available by Chogon. Hoikas has ruminated on this matter in the Discord Lake Lighting thread, and further programming input/assistance on the topic is welcomed.

Lastly, there's been talk about having lake light levels affect the overall lighting in Ae'gura. Most of the lighting in the City is baked into the meshes and cannot be changed one day to the next, but these interactive lighting changes would require Plasma runtime lights. We're not sure we want to open what could become a can or worms, in which such lighting would have to be specifically assigned to some aspects of the City but excluded from others. Additionally, some parts of Ae'gura were never designed to be seen at substantially brighter levels than we've always seen them. So for now at least, we'd prefer to set aside this issue that could become a magillah.

EhrenCG commented:
>     * Fog settings will be minimized in order to avoid light pollution on the backdrop image

It just occurred to me a little while ago, shouldn't there be a way to make the backdrop ignore the fog?  This one sure does (ignoring the GZ cube):
![HoodFog](https://user-images.githubusercontent.com/10370754/142756161-7b76eef7-1e9b-4c02-8fc3-5367f20e1528.png)

No idea how it actually works or its limits though, heh.  I just find it interesting, Kadish's small slice of sky is like that too and for that matter it's how the Minkata stars are still visible in the thickest fog.

Emor-Dni-Lap commented:
Ehren, I admit I haven't opened a 'hood file in prpshop to check fog settings there, nor in say K'veer or the GZ.
But in the Neighborhood there is just no evidence of visible fog anywhere, to my eyeballs at least. So if there **is** fog established in a 'hood's settings, it seems likely to be set at some very negligible level.

Hazado commented:
The fog in the hood is basically turned off. Though I did find the settings that makes the textures ignore the fog. Thank you for that Ehren!
